app/assets/javascripts/components/google_maps_autocomplete_input.es6.jsx

```diff
@@ -0,0 +1,123 @@
+class GoogleMapsAutocompleteInput extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.$addressInput = $($(`#${this.props.addressInputId}`));
+    this.$latitudeInput = $($(`#${this.props.latitudeInputId}`));
+    this.$longitudeInput = $($(`#${this.props.longitudeInputId}`));
+
+    if (this.props.defaultLocation && this.props.defaultLocation.lat) {
+      this.defaultZoom = 15;
+      this.defaultLocation = this.props.defaultLocation;
+    } else {
+      this.defaultZoom = 12;
+      this.defaultLocation = { lat: 41.3833, lng: 2.1833 };
+    }
+  }
+
+  componentDidMount() {
+    $(document).on('gmaps:loaded', (event) => this.forceUpdate());
```

Comment:
Can you be sure this got called? Maybe the react component got initialized after the event was triggered. What if we're loading the component asynchronously?

I often solve this with a promise - it resolves when GMaps loads.